src/movepick.cpp

```diff
@@ -156,16 +153,11 @@ Move MovePicker::next_move(bool skipQuiets) {
 top:
   switch (stage) {
 
-  case MAIN_TT:
-  case EVASION_TT:
-  case QSEARCH_TT:
-  case PROBCUT_TT:
-      ++stage;
-      return ttMove;
-
   case CAPTURE_INIT:
   case PROBCUT_INIT:
   case QCAPTURE_INIT:
+      if (!(useTTM = !useTTM)) return ttMove; //if set, unset and return ttm
```

Comment:
new line for return.
can't we rewrite that '(!(useTTM = !useTTM)) ' more cleanly, it is hard to parse.

Reply:
It can be more clear, but will probably require brackets like:
if (useTTM)
{
     useTTM = !useTTM;
     return ttMove;
}

Is that better?

Reply:
easier to read... 

makes one wonder if this is really a simplification.

Reply:
I could also do a simple method that does the same.  it would make the jump table quite easy to read, but a few more lines.  To me, this is still better (more simple) that a bunch of TT stages.

Reply:
I added a new line after the check and added a comment.  To me, this seems pretty easy to read/understand.